basics/getting_started/plugin_compatibility.md

```diff
@@ -2,80 +2,102 @@
+
+>**Note** By convention, all plugins should declare a dependency on **`com.intellij.modules.lang`**. 
+
+| Plugin Dependency:<br>Module or Built-in Plugin  | Functionality  | IntelliJ Platform-Based Products  | 
+|--------------------------------------------------|----------------|:---------------------------------:|
+| `com.intellij.modules.platform`  | Messaging, UI Themes, UI Components, Files, Documents, Actions, Components, Services, Extensions, Editors  | All      |
+| **`com.intellij.modules.lang`**  | File Type, Lexer, Parser, Highlighting, References, Code Completion, Find, Rename, Formatter, Code Navigation  | All  |
+| `com.intellij.modules.xml`       | XML, XML DOM, XSD/DTD, DOM Model  | All                                                                               |
+| `com.intellij.modules.vcs`       | VCS Revision Numbers, File Status, Change Lists, File History, Annotations  | All                                     |
+| `com.intellij.modules.xdebugger` | Debug Session, Stack Frames, Break Points, Source Positions, Memory Views, Tracked Instances | All                    |
+
+For example, if a plugin is dependent _only_ on one or more of the modules in the table above, and declares the module dependencies in `plugin.xml`, then any product based on the IntelliJ Platform will load it.
+
+### Modules Specific to Functionality
+More specialized functionality is also delivered via modules in IntelliJ Platform-based products. 
+For example, the `com.intellij.modules.java` module supports the language-specific Java functionality. 
+If a plugin uses functionality from this module, such as Java-specific inspections and refactoring, it must declare a dependency on this module.
+
+Note that not all products define and declare modules. 
+For example, PhpStorm does not have its own module, but the product itself depends on (and ships with) the PHP language plugin. 
+A plugin project is compatible with PHP functionality if it declares a dependency on this PHP language plugin.
+
+The following table lists modules or built-in plugins that provide specific functionality, and the products that currently ship with them. 
+Module FQN is listed on the left, an overview of functionality is listed in the middle column, and availability in products is listed on the right:
+
+| Plugin Dependency:<br>Module or Built-in Plugin  | Functionality  | IntelliJ Platform-Based Products  | 
+|--------------------------------------------------|----------------|-----------------------------------|
+| `com.intellij.modules.java`          | **Java** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework  | IntelliJ IDEA, Android Studio         |
+| `com.intellij.modules.ultimate`      | JavaScript and TypeScript language PSI Models, Java EE, JVM Frameworks: Spring, Play, Grails...  | IntelliJ IDEA Ultimate Edition      |
```

Comment:
for `ultimate`, maybe change description to "all functionality available in IJ Ultimate only"